### ISSUE TITLE
typescript declaration file

### ISSUE THREAD

**Issue by JasinYip** [samirkumardas/jmuxer#74]
Thank you so much for implemetation of JMuxer. It's a great project!

I'm using JMuxer for a typescript-based project which shows the following error:

```
Could not find a declaration file for module 'jmuxer'. '/Users/jasinyip/prj/driving_test_pc_ui/node_modules/jmuxer/dist/jmuxer.min.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/jmuxer` if it exists or add a new declaration (.d.ts) file containing `declare module 'jmuxer';`ts(7016)
```

Could you please offer a typescription declaration file for TypeScript projects? Thank you so much!

**Comment by Lillifee** [comment]
Yeah, it would be nice to have the type definition out of the box.
Maybe someone more familiar with the library can add it to DefinitelyTyped?

I started a .d.ts file for my project:
https://github.com/Lillifee/raspiCam

btw. Great library and thanks a lot!

**types/jmuxer.d.ts**
```
declare module 'jmuxer' {
  interface JMuxerOptions {
    /**
     * String ID of a video tag / Reference of the HTMLVideoElement.
     * Required field for browsers.
     */
    node: string;

    /**
     * Available values are: both, video and audio.
     * Default is both
     */
    mode?: 'both' | 'video' | 'audio';

    /**
     * Buffer flushing time in seconds.
     * Default value is 1500 miliseconds.
     */
    flushingTime?: number;

    /**
     * Either it will clear played media buffer automatically or not. Default is true.
     */
    clearBuffer?: boolean;

    /**
     * Frame rate of the video if it is known/fixed value.
     * It will be used to find frame duration if chunk duration is not available with provided media data.
     */
    fps?: number;

    /**
     * Will be called once MSE is ready.
     */
    onReady?: () => void;

    /**
     * Will print debug log in browser console. Default is false.
     */
    debug?: boolean;
  }

  interface JMuxerFeedArgs {
    /**
     * autio - AAC buffer
     */
    audio?: Uint8Array;

    /**
     * video - h264 buffer
     */
    video?: Uint8Array;

    /**
     * duration in miliseconds of the provided chunk.
     * If duration is not provided, it will calculate frame duration wtih the provided frame rate (fps).
     */
    duration?: number;
  }

  declare class JMuxer {
    /**
     * Creates an instance of JMuxer.
     */
    constructor(options: JMuxerOptions);

    /**
     * Object properites may have audio, video and duration.
     * At least one media property i.e audio or video must be provided.
     * If no duration is provided, it will calculate duration based on fps value
     */
    feed(args: JMuxerFeedArgs): void;

    /**
     * Destroy the jmuxer instance and release the resources
     */
    destroy(): void;
  }

  export = JMuxer;
}
```

**Comment by samirkumardas** [comment]
@Lillifee  I have added the declaration file for jmuxer and requested PR in https://github.com/DefinitelyTyped/DefinitelyTyped
If you have some time, could you please review and approve it if everything is okay? https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52634/files
Please let me know if you have any suggestions.

**Comment by Lillifee** [comment]
I'm not sure if we can improve the return type of the createStream. I added a comment but feel free to close it if you can't find a quick solution. 

Let me know if you need a additional approval to merge it back to master.
Thanks a lot and good job!

**Comment by samirkumardas** [comment]
@Lillifee Thank you for your feedback. I tried Duplex earlier as well but did not work. But the comment `/// <reference types="node" />` did the trick. It is working fine now. You need to review it again for this new change.

**Comment by Lillifee** [comment]
I approved it. Thanks a lot for your effort!

**Comment by samirkumardas** [comment]
@Lillifee  It is available on npmjs https://www.npmjs.com/package/@types/jmuxer
Thank you for your effort and time!
